### PR TITLE
Added SendingCommand event

### PR DESCRIPTION
**Pull request by angelseraphim**
:3

### CODE REVIEW

File: Exiled.Events/EventArgs/Player/SendingCommandEventArgs.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+        /// <summary>
+        /// Gets command interface.
+        /// </summary>
+        public ICommand Command { get; }
+
+        /// <summary>
+        /// Gets querry.
+        /// </summary>
+        public string Querry { get; }
```
Comment (xNexusACS) [inline Exiled.Events/EventArgs/Player/SendingCommandEventArgs.cs:56]: Typo: Query

File: Exiled.Events/EventArgs/Player/SendingCommandEventArgs.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        /// <param name="player">
+        /// <inheritdoc cref="Player" />
+        /// </param>
+        /// <param name="command">
+        /// <inheritdoc cref="Command" />
+        /// </param>
+        /// <param name="querry">
+        /// <inheritdoc cref="Querry" />
+        /// </param>
+        public SendingCommandEventArgs(Player player, ICommand command, string querry)
```
Comment (xNexusACS) [inline Exiled.Events/EventArgs/Player/SendingCommandEventArgs.cs:36]: Typo: Query

File: Exiled.Events/Patches/Events/Player/SendingCommand.cs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+                    // Sender
+                    new CodeInstruction(OpCodes.Ldarg_1),
+
+                    // Player.Get(CommandSender)
+                    new CodeInstruction(OpCodes.Call, Method(typeof(Player), nameof(Player.Get), new Type[] { typeof(CommandSender) })),
+
+                    // command
+                    new CodeInstruction(OpCodes.Ldloc_1),
+
+                    // querry
```
Comment (xNexusACS) [inline Exiled.Events/Patches/Events/Player/SendingCommand.cs:61]: Typo: Query

File: Exiled.Events/Patches/Events/Player/SendingCommand.cs [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+                    new CodeInstruction(OpCodes.Ldarg_0),
+
+                    // SendingCommandEventArgs ev = new SendingCommandEventArgs(Player, ICommand, Querry)
```
Comment (xNexusACS) [inline Exiled.Events/Patches/Events/Player/SendingCommand.cs:64]: Typo: Query

**Review by xNexusACS — CHANGES_REQUESTED** [review]

File: Exiled.Events/Patches/Events/Player/SendingCommand.cs [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+    using RemoteAdmin;
+
+    using static HarmonyLib.AccessTools;
+
+    /// <summary>
+    /// Patches <see cref="CommandProcessor.ProcessQuery(string, CommandSender)" />.
+    /// Adds the <see cref="Handlers.Player.SendingCommand" /> event.
+    /// </summary>
+    ///
```
Comment (xNexusACS) [inline Exiled.Events/Patches/Events/Player/SendingCommand.cs:32]: unnecesary ///

File: Exiled.Events/EventArgs/Player/SendingCommandEventArgs.cs [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        /// </summary>
+        public bool IsAllowed { get; set; } = true;
+
+        /// <summary>
+        /// Gets command interface.
+        /// </summary>
+        public ICommand Command { get; }
+
+        /// <summary>
+        /// Gets query.
```
Comment (xNexusACS) [inline Exiled.Events/EventArgs/Player/SendingCommandEventArgs.cs:54]: Gets the query of the command.

File: Exiled.Events/EventArgs/Player/SendingCommandEventArgs.cs [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+            Query = query;
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the command can be executed.
+        /// </summary>
+        public bool IsAllowed { get; set; } = true;
+
+        /// <summary>
+        /// Gets command interface.
```
Comment (xNexusACS) [inline Exiled.Events/EventArgs/Player/SendingCommandEventArgs.cs:49]: Gets the command that is being executed.

**Review by xNexusACS — CHANGES_REQUESTED** [review]
lets improve a little bit some of the summaries

**Comment by xNexusACS** [comment]
one question I have, does either Query or Command becomes null when the command doesn't exists?

**Comment by angelseraphim** [comment]
The patch works after the command. Because of this, it cannot be null

**Comment by xNexusACS** [comment]
> The patch works after the command. Because of this, it cannot be null

what I mean is, if the command doesn't exist in RemoteAdmin, does the ICommand inferface becomes null?

**Comment by angelseraphim** [comment]
If there is no command, the event will not be called
CommandProcessor.RemoteAdminCommandHandler.TryGetCommand

**Comment by xNexusACS** [comment]
> If there is no command, the event will not be called CommandProcessor.RemoteAdminCommandHandler.TryGetCommand

oh, nice

**Review by xNexusACS — APPROVED** [review]

**Comment by angelseraphim** [comment]
:D

**Pull request merged**